Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -55,9 +57,13 @@ class CallUIAdapter {
             Logger.info("\(TAG) choosing non-callkit adaptee for older iOS")
             adaptee = NonCallKitCallUIAdaptee(callService: callService, notificationsAdapter: notificationsAdapter)
         }
+
+        audioService = CallAudioService(handleRinging: adaptee.hasManualRinger)
     }
 
     internal func reportIncomingCall(_ call: SignalCall, thread: TSContactThread) {
+        call.addObserverAndSyncState(observer: audioService)
```

Comment:
What will happen if user receives _two_ incoming calls before accepting either?  Could ring state end up in a bad state?